test/parallel/test-http-pipeline-flood.js

```diff
@@ -78,6 +78,6 @@ function child() {
   conn.on('drain', common.mustCall(write));
 
   function write() {
-    while (false !== conn.write(req, 'ascii'));
+    while (conn.write(req, 'ascii'));
```

Comment:
Dropping the strict comparison is a semantic change that I'm not sure is actually correct.  Same comment applies to a couple of other places.

Reply:
Write operations should always either return `true` or `false`. If that is not the case, it would be an error.

For read operations it returns a string, a buffer or null. The latter when there is nothing to read. Buffer is always truthy and the string should (to my best knowledge) always be non empty. I might be wrong about the empty string case though. But here in our tests we definitely do not have any empty strings.

Reply:
> Write operations should always either return true or false.

In the past we treated (for example) `undefined` different from `false`.  In fact, I suspect the change to `lib/internal/streams/legacy.js` is a subtle regression that just happens to fly under the radar of our test suite.